openhands/storage/settings_store.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import annotations
```

Comment:
I understand the desire for an abstract class, but maybe for simplicity we just have the file implementation, and then any overriders inherit from that? Then we can just call it SettingsStore

Reply:
Also maybe we put these in a new dir, `openhands/storage/settings/...`